modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -21,14 +21,45 @@
 @{
     @defgroup gapi_filters Graph API: Image filters
     @defgroup gapi_colorconvert Graph API: Converting image from one color space to another
+    @defgroup gapi_feature Graph API: Image Feature Detection
+    @defgroup gapi_shape Graph API: Image Structural Analysis and Shape Descriptors
 @}
  */
 
+namespace {
+void checkMetaForFindingContours(const int depth, const int chan, const int mode)
+{
+    GAPI_Assert(chan == 1);
+    switch (mode)
+    {
+    case cv::RETR_CCOMP:
+        GAPI_Assert(depth == CV_8U || depth == CV_32S);
+        break;
+    case cv::RETR_FLOODFILL:
+        GAPI_Assert(depth == CV_32S);
+        break;
+    default:
+        GAPI_Assert(depth == CV_8U);
+        break;
+    }
+}
+
+// Checks if the passed mat is a set of n-dimentional points of the given depth
+bool isPointsVector(const int chan, const cv::Size &size, const int depth,
```

Comment:
> isPointsVector

Why "points"?

Refer to OpenCV functionality. It is called `checkVector`. Also you may want to use `checkArray()` naming here.

Reply:
@alalek This checks if it's vector of points represented via `GMatDesc`. Although, I agree that this naming is debatable.
Don't want to refer fully to `checkVector() ` method as this is a cut version of it, not fully corresponding to all the functionality of the original `Mat::checkVector()`.
Do you insist to rename to `checkArray()`?

Reply:
this name works well, given its intent, I believe